File: Data Structures & Algorithms/top-k-elements-in-list/submission-8.cpp
class Solution
{
public:
    vector<int> topKFrequent(vector<int> &nums, int k)
    {
        unordered_map<int, int> map;
        vector<vector<int> > arr(nums.size() + 1);
        for (int num: nums)
        {
            map[num]++;
        }
        for (auto const &it: map)
        {
            arr[it.second].push_back(it.first);
        }
        vector<int> res;
        for (int i = arr.size() - 1; i >= 0; i--)
        {
            if (arr[i].size() > 0)
            {
                for (int j: arr[i])
                {
                    res.push_back(j);
                    if (res.size() == k) return res;
                }
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-8.java
class Solution
{
    public int[] productExceptSelf(int[] nums)
    {
        int[] arr = new int[nums.length];
        Arrays.fill(arr, 1);
        for (int i = 0; i < nums.length; i++)
        {
            for (int j = 0; j < i; j++)
            {
                arr[i] *= nums[j];
            }
            for (int j = i + 1; j < nums.length; j++)
            {
                arr[i] *= nums[j];
            }
        }

        return arr;
    }
}